src/native/corehost/comhost/comhost.cpp

```diff
@@ -431,14 +430,14 @@ namespace
 
             regKey.reset(regProgIdKeyRaw);
 
-            // The default value for the key is the ProgID
+            // The default value for the key is the type - this matches RegAsm behavior.
```

Comment:
We ignore what the user has specified as the ProgID and use the type name to set `SOFTWARE\Classes\CLSID\<clsid>\ProgID`, but add a key under `SOFTWARE\Classes\` using the specified ProgID? Seems odd to me - is this seriously what RegAsm does?

Reply:
...

Happy new year!